packages/maskbook/src/components/GuideStep/index.tsx

```diff
@@ -0,0 +1,266 @@
+    background: '#1C68F3',
+})
+
+const usernameTheme = extendsTheme((theme: Theme) => {
+    const color = theme.palette.mode === 'light' ? '#fff' : '#0f1419'
+
+    return {
+        components: {
+            MuiInputLabel: {
+                styleOverrides: {
+                    root: {
+                        color,
+                        '&.Mui-disabled': { color },
+                    },
+                },
+            },
+            MuiInputAdornment: {
+                styleOverrides: {
+                    root: {
+                        color,
+                    },
+                },
+            },
+            MuiOutlinedInput: {
+                styleOverrides: {
+                    input: {
+                        color,
+                        ':disabled': { color, '-webkit-text-fill-color': color },
+                    },
+                },
+            },
+            MuiTextField: {
+                styleOverrides: {
+                    root: {
+                        color,
+                        fieldset: {
+                            borderColor: `${color} !important`,
+                        },
+                    },
+                },
+            },
+        },
+    }
+})
+
+const FindUsername = () => {
+    const { t } = useI18N()
+    const username = localStorage.getItem('username')
+
+    return (
+        <>
+            <Typography variant="h3" sx={{ fontSize: '18px' }}>
+                {t('setup_guide_find_username_title')}
+            </Typography>
+            <ThemeProvider theme={usernameTheme}>
+                <TextField
+                    disabled
+                    fullWidth
+                    label={t('username')}
+                    value={username}
+                    sx={{ margin: '24px 0 16px' }}
+                    InputProps={{
+                        size: 'small',
+                        startAdornment: (
+                            <InputAdornment position="start">
+                                <AlternateEmailIcon sx={{ fontSize: '16px' }} />
+                            </InputAdornment>
+                        ),
+                    }}
+                />
+            </ThemeProvider>
+        </>
+    )
+}
+
+export interface GuideStepProps extends PropsWithChildren<{}> {
+    total: number
+    step: number
+    tip: string
+    arrow?: boolean
+    onComplete?: () => void
+}
+
+export default function GuideStep({ total, step, tip, children, arrow = true, onComplete }: GuideStepProps) {
+    const { t } = useI18N()
+    const { classes } = useStyles()
+    const childrenRef = useRef<HTMLElement>()
+    const [clientRect, setClientRect] = useState<any>({})
+    const [open, setOpen] = useState(false)
+    const ui = activatedSocialNetworkUI
+    const lastStepRef = currentSetupGuideStatus[ui.networkIdentifier]
+    const lastStep = useValueRef(lastStepRef)
+
+    useEffect(() => {
+        setOpen(+lastStep === step)
+    }, [lastStep])
```

Comment:
Just for double confirmation: it's on purpose to exclude `step` from the deps list.